apps/teslafi/teslafi.star

```diff
@@ -0,0 +1,360 @@
+""")
+
+def lightness(color, amount):
+    hsl_color = rgb_to_hsl(*hex_to_rgb(color))
+    hsl_color_list = list(hsl_color)
+    hsl_color_list[2] = hsl_color_list[2] * amount
+    hsl_color = tuple(hsl_color_list)
+    return rgb_to_hex(*hsl_to_rgb(*hsl_color))
+
+def rgb_to_hsl(r, g, b):
+    r = float(r / 255)
+    g = float(g / 255)
+    b = float(b / 255)
+    high = max(r, g, b)
+    low = min(r, g, b)
+    h, s, l = ((high + low) / 2,) * 3
+
+    if high == low:
+        h = 0.0
+        s = 0.0
+    else:
+        d = high - low
+        s = d / (2 - high - low) if l > 0.5 else d / (high + low)
+        if high == r:
+            h = (g - b) / d + (6 if g < b else 0)
+        elif high == g:
+            h = (b - r) / d + 2
+        elif high == b:
+            h = (r - g) / d + 4
+        h /= 6
+
+    return int(math.round(h * 360)), s, l
+
+def hsl_to_rgb(h, s, l):
+    def hue_to_rgb(p, q, t):
+        if t < 0:
+            t += 1
+        if t > 1:
+            t -= 1
+        if t < 1 / 6:
+            return p + (q - p) * 6 * t
+        if t < 1 / 2:
+            return q
+        if t < 2 / 3:
+            return p + (q - p) * (2 / 3 - t) * 6
+        return p
+
+    h = h / 360
+    if s == 0:
+        r, g, b = (l,) * 3  # achromatic
+    else:
+        q = l * (1 + s) if l < 0.5 else l + s - l * s
+        p = 2 * l - q
+        r = hue_to_rgb(p, q, h + 1 / 3)
+        g = hue_to_rgb(p, q, h)
+        b = hue_to_rgb(p, q, h - 1 / 3)
+
+    return int(math.round(r * 255)), int(math.round(g * 255)), int(math.round(b * 255))
+
+def hex_to_rgb(color):
+    # Expand 4 digit hex to 7 digit hex
+    if len(color) == 4:
+        x = "([A-Fa-f0-9])"
+        matches = re.match("#%s%s%s" % (x, x, x), color)
+        rgb_hex_list = list(matches[0])
+        rgb_hex_list.pop(0)
+        for i in range(len(rgb_hex_list)):
+            rgb_hex_list[i] = rgb_hex_list[i] + rgb_hex_list[i]
+        color = "#" + "".join(rgb_hex_list)
+
+    # Split hex into RGB
+    x = "([A-Fa-f0-9]{2})"
+    matches = re.match("#%s%s%s" % (x, x, x), color)
+    rgb_hex_list = list(matches[0])
+    rgb_hex_list.pop(0)
+    for i in range(len(rgb_hex_list)):
+        rgb_hex_list[i] = int(rgb_hex_list[i], 16)
+    rgb = tuple(rgb_hex_list)
+
+    return rgb
+
+# Convert RGB tuple to hex
+def rgb_to_hex(r, g, b):
+    return "#" + str("%x" % ((1 << 24) + (r << 16) + (g << 8) + b))[1:]
+
+def main(config):
+    if config.str("api") == None:
+        return render.Root(
+            child = render.WrappedText("Please enter the API info!"),
+        )
+    else:
+        rep_cache = cache.get("teslafi")
```

Comment:
I think you want to include the api key in the cache key. The cache is shared across all instances of this app (but not with other apps), so the "teslafi" key will be written to (and read from) on every run otherwise, effectively showing you someone else's data most of the time.